File: backend/app/schemas/webdav.py
"""Pydantic models for WebDAV API responses."""

from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class WebdavStatusResponse(BaseModel):
    """Detailed WebDAV server status (admin only)."""
    is_running: bool
    port: int
    ssl_enabled: bool
    started_at: Optional[datetime] = None
    worker_pid: Optional[int] = None
    last_heartbeat: Optional[datetime] = None
    error_message: Optional[str] = None
    connection_url: Optional[str] = None


class OsConnectionInfo(BaseModel):
    """Mount instructions for a specific OS."""
    os: str
    label: str
    command: str
    notes: Optional[str] = None


class WebdavConnectionInfo(BaseModel):
    """OS-specific mount instructions with personalised username."""
    is_running: bool
    port: int
    ssl_enabled: bool
    username: str
    connection_url: str
    instructions: list[OsConnectionInfo]

File: client/src/api/webdav.ts
import { apiClient } from '../lib/api';

export interface WebdavStatus {
  is_running: boolean;
  port: number;
  ssl_enabled: boolean;
  started_at: string | null;
  worker_pid: number | null;
  last_heartbeat: string | null;
  error_message: string | null;
  connection_url: string | null;
}

export interface OsConnectionInfo {
  os: string;
  label: string;
  command: string;
  notes: string | null;
}

export interface WebdavConnectionInfo {
  is_running: boolean;
  port: number;
  ssl_enabled: boolean;
  username: string;
  connection_url: string;
  instructions: OsConnectionInfo[];
}

export async function getWebdavStatus(): Promise<WebdavStatus> {
  const { data } = await apiClient.get<WebdavStatus>('/api/webdav/status');
  return data;
}

export async function getWebdavConnectionInfo(): Promise<WebdavConnectionInfo> {
  const { data } = await apiClient.get<WebdavConnectionInfo>('/api/webdav/connection-info');
  return data;
}
